app/src/main/java/org/wikipedia/analytics/metricsplatform/ArticleEvent.kt

```diff
@@ -219,10 +227,15 @@ class ArticleLinkPreviewInteraction : TimedMetricsEvent {
     private fun submitEvent(action: String) {
         submitEvent(
             "article_link_preview_interaction",
```

Comment:
Hi @cjming 
The updates look good to me... just have a couple questions:
In the new way of wiring the events, I see that we are no more committing them to a specific stream. Does this mean we will not have an option to have multiple streams to the same schema? I ask this because, this is the easiest way for @Shay to work with a generic schema for multiple features. 
However, I see that the schema name, event name and action and actionSubtype distinction which might help with the same.  Just wanted to understand better, the role of streams here.

Reply:
hi @sharvaniharan - thanks for the Qs - responses inline below:

> Does this mean we will not have an option to have multiple streams to the same schema?

Multiple streams can be defined with the same schema in this context -- this will have to be done in config which would need to be in production before events can be submitted/validated/accepted. 

So for example the [MP base app schema](https://gerrit.wikimedia.org/r/plugins/gitiles/schemas/event/secondary/+/refs/heads/master/jsonschema/analytics/product_metrics/app/base/current.yaml) can be defined as the schema for multiple stream names as long as there isn't any need for custom data. Here's an example:

```
	'android.example_stream1' => [
		'schema_title' => 'analytics/product_metrics/app/web_base',
		'destination_event_service' => 'eventgate-analytics-external',
		'producers' => [
			'metrics_platform_client' => [
				'provide_values' => [
                                          'page_namespace_name,
                                          'performer_is_temp',
					'performer_groups',
				],
			],
		],
	],
	'android.example_stream2' => [
		'schema_title' => 'analytics/product_metrics/app/web_base',
		'destination_event_service' => 'eventgate-analytics-external',
		'producers' => [
			'metrics_platform_client' => [
				'provide_values' => [
                                          'page_id',
                                          'page_namespace_id,
                                          'performer_is_logged_in',
					'performer_session_id',
				],
			],
		],
	],
```
In the above examples, multiple streams can have events validated/accepted by the same base schema and even customize the contextual values submitted with their events.

If custom data is needed however, then there will most likely be a 1-to-1 relationship between stream name and custom schema in config.

Let me know if this isn't clear of if you have more questions.